lib/TH/generic/THStorage.c

```diff
@@ -107,9 +107,9 @@ void THStorage_(free)(THStorage *storage)
   if(!storage)
     return;
 
-  if((storage->flag & TH_STORAGE_REFCOUNTED) && (storage->refcount > 0))
+  if((storage->flag & TH_STORAGE_REFCOUNTED) && (THAtomicGet(&storage->refcount) > 0))
   {
```

Comment:
On Friday, April 10, 2015 10:44:05 AM Ronan Collobert wrote:
> i hesitated to access atomic variables only with atomic functions, as an
> habit. (in fact, my first version had atomic as a struct, to refrain the
> user touching directly the number in there). true, here does not do much...
> i put it only for consistency reasons. you think i should remove it?

When I started using atomics in djvu, I decided not to use atomic gets and 
sets on aligned elementary types (1,2,4,8 bytes longs things) because I do not 
believe that reasonable hardware would break such aligned accesses into 
multiple parts. I only use them for basic read-modify-write operations such as 
increment, decrement, compare-and-swap, test-and-set, etc.

It is possible to be much more refined in using atomics. One can go faster by 
adequately using acquire/release semantics to ensure correct memory ordering. 
But I never felt comfortable with these things and never got a good platform 
to test them (the intel platform is very forgiving of these). So I stay with a 
few basic read-modify-write operations...

- L.